docs/notebooks/tutorials/adv_newcv_subclass.ipynb

```diff
@@ -0,0 +1,509 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def sigmoid(x,sigma,a,b):\n",
+    "    return 1 - (1+(2**(a/b) -1)*( x/sigma)) **(-b/a)\n",
+    "\n",
+    "class SketchMapLoss(torch.nn.Module):\n",
+    "    \"\"\"Sketch-map loss function\"\"\"\n",
+    "\n",
+    "    def __init__(self, high_dim_params = None, low_dim_params = None, **kwargs) -> None:\n",
+    "        \"\"\"Constructor.\n",
+    "\n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        high_dim_params : dict, optional\n",
+    "            parameters for sigmoid acting on high-dim data, by default dict(sigma=0.75, a=12, b=6)\n",
+    "        low_dim_params : dict, optional\n",
+    "            parameters for sigmoid acting on low-dim data,  by default dict(sigma=1, a=2, b=6)\n",
+    "        \"\"\"\n",
+    "        super().__init__(**kwargs)\n",
+    "    \n",
+    "        # Parameters for sigmoid functions (default values)\n",
+    "        self.high_dim_params = high_dim_params if high_dim_params is not None else dict(sigma=0.75, a=12, b=6)\n",
+    "        self.low_dim_params = low_dim_params if low_dim_params is not None else dict(sigma=1, a=2, b=6)\n",
+    "\n",
+    "    def forward(self, X : torch.Tensor , s : torch.Tensor) -> torch.Tensor:\n",
+    "        \"\"\"Compute\n",
+    "\n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        X : torch.Tensor\n",
+    "            high-dim points\n",
+    "        s : torch.Tensor\n",
+    "            low-dim points\n",
+    "\n",
+    "        Returns\n",
+    "        -------\n",
+    "        torch.Tensor\n",
+    "            scalar loss \n",
+    "        \"\"\"\n",
+    "\n",
+    "        # compute pairwise distances\n",
+    "        X_ij = torch.cdist(X,X)\n",
+    "        s_ij = torch.cdist(s,s)\n",
+    "        n_dist = (X.shape[0]-1)**2\n",
+    "\n",
+    "        # compute sketch-map cost\n",
+    "        loss = 1/n_dist * torch.sum( ( sigmoid(X_ij, **self.high_dim_params ) - sigmoid(s_ij, **self.low_dim_params ) )**2 ) \n",
+    "\n",
+    "        return loss"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining the CV"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now define an `EncoderMap` CV by subclassing the `AutoEncoderCV` class. This allows us to retain all of the methods already implemented for the autoencoder and just add the sketch-map loss. \n",
+    "\n",
+    "**\\_\\_init__**\n",
+    "\n",
+    "The constructor, besides the usual arguments of `AutoEncoderCV` which are repeated for convenience, takes the following arguments:\n",
+    "- `ratio_losses` which set the relative weighting of the sketch-map/reconstruction loss.\n",
```

Comment:
``losses_ratio`` sounds a bit better to my ear

Reply:
ok, will change it!